mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -848,6 +850,11 @@ public void applyRightsCommand(MailboxPath mailboxPath, MailboxACL.ACLCommand ma
     @Override
     public void setRights(MailboxPath mailboxPath, MailboxACL mailboxACL, MailboxSession session) throws MailboxException {
         storeRightManager.setRights(mailboxPath, mailboxACL, session);
+
+        mailboxACL.getEntries()
+            .keySet()
+            .stream()
+            .forEach(entryKey -> dispatcher.aclLookupAdded(session, mailboxPath, entryKey));
     }
```

Comment:
What if.... the stored ACL does not match the one that was passed here?

Reply:
This comment should be fix by pass only aclDiff